src/inmanta/protocol/rest/__init__.py

```diff
@@ -183,52 +202,61 @@ async def process(self) -> None:
-                            value = [self._message[arg]]
-                        else:
-                            value = self._message[arg]
-                        all_fields.remove(arg)
-                    # Pre-process dict params for GET
-                    elif arg_type and self._properties.operation == "GET" and self._is_dict_or_optional_dict(arg_type):
-                        dict_prefix = f"{arg}."
-                        dict_with_prefixed_names = {
-                            param_name: param_value
-                            for param_name, param_value in self._message.items()
-                            if param_name.startswith(dict_prefix) and len(param_name) > len(dict_prefix)
-                        }
-                        value = (
-                            await self._get_dict_value_from_message(arg, dict_prefix, dict_with_prefixed_names)
-                            if dict_with_prefixed_names
-                            else None
-                        )
-
-                        for key in dict_with_prefixed_names.keys():
-                            all_fields.remove(key)
-
-                    else:  # get default value
-                        value = self.get_default_value(arg, i, defaults_start)
-                else:  # get default value
+            arg_type: Optional[Type[object]] = self._argspec.annotations.get(arg)
+            if arg in self._message:
+                # Argument is parameter in body of path of HTTP request
+                if (
+                    arg_type
+                    and self._properties.operation == "GET"
+                    and typing_inspect.is_generic_type(arg_type)
+                    and issubclass(typing_inspect.get_origin(arg_type), list)
+                    and not isinstance(self._message[arg], list)
+                    and len(typing_inspect.get_args(arg_type, evaluate=True)) == 1
+                    and isinstance(self._message[arg], typing_inspect.get_args(arg_type)[0])
+                ):
+                    # If a GET endpoint has a parameter of type list that is encoded as a URL query parameter and the
+                    # specific request provides a list with one element, urllib doesn't parse it as a list.
+                    # Map it here explicitly to a list.
+                    value = [self._message[arg]]
+                else:
+                    value = self._message[arg]
+                all_fields.remove(arg)
+            elif arg_type and self._properties.operation == "GET" and self._is_dict_or_optional_dict(arg_type):
+                # Argument is dictionary-based expression in query parameters of GET operation
+                dict_prefix = f"{arg}."
+                dict_with_prefixed_names = {
+                    param_name: param_value
+                    for param_name, param_value in self._message.items()
+                    if param_name.startswith(dict_prefix) and len(param_name) > len(dict_prefix)
```

Comment:
>  len(param_name) > len(dict_prefix)
I'm curious, what is this for ? To prevent matching exactly f"{arg}." with no attribute behind it ?

Reply:
yes